extensions/sql-migration/src/scripts/SHIR-auto-configuration.ps1

```diff
@@ -0,0 +1,609 @@
+			Register-IntegrationRuntime -AuthKey1 $AuthKey1 -AuthKey2 $AuthKey2
+		}
+	}
+}
+
+Function Get-LogFilePath {
+	<#
+    .DESCRIPTION
+    Creates the log file path inside the user profile and returns the path.
+
+    .OUTPUTS
+    Log file path
+    #>
+
+	# Get the current user's home directory using the environment variable USERPROFILE
+	$home_dir = $env:USERPROFILE
+
+	# Create the .shir folder inside the home directory using the Join-Path cmdlet
+	$shir_dir = Join-Path $home_dir "shir"
+	New-Item -ItemType Directory -Path $shir_dir -Force
+
+	# Create the logs.txt file inside the .shir folder using the New-Item cmdlet
+	$logs_file = Join-Path $shir_dir "logs.txt"
+	New-Item -ItemType File -Path $logs_file -Force
+
+	Write-OutputAndLog ("Log file generated: " + $logs_file)
+
+	return $logs_file
+}
+
+################################### Pre-installation checks ##################################
+Function Test-CanInstallIR {
+	<#
+    .DESCRIPTION
+    Performs pre-installation checks to ensure that SHIR can be installed, if any check fails a terminating error is thrown.
+
+    .OUTPUTS
+    [bool]: True if pre-installation validation is passed else false.
+    #>
+	try {
+		if (-not [Environment]::Is64BitOperatingSystem) {
+			throw "Pre-requisite not met - OS is not 64 bits."
+		}
+		if (-not (Test-DotNetVersion)) {
+			throw "Pre-requisite not met - Dotnet version is lesser than 4.7.2"
+		}
+		if ([string]::IsNullOrWhitespace($AuthKey1) -and [string]::IsNullOrWhitespace($AuthKey2)) {
+			throw "You need to provide atleast one authentication key."
+		}
+	}
+	catch {
+		Write-ErrorAndLog -exception $_.Exception
+		return $false
+	}
+	return $true
+}
+
+Function Test-DotNetVersion {
+	<#
+    .DESCRIPTION
+    Checks if installed .NET Framework version >= 4.7.2.
+    https://learn.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed
+
+    .OUTPUTS
+    [bool]: true if intalled .NET Framework version >= 4.7.2 else false
+    #>
+	return (Get-ItemPropertyValue -LiteralPath 'HKLM:SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full' -Name Release) -ge 461808
+}
+
+Function Get-InstalledShirVersion {
+	<#
+    .DESCRIPTION
+    Gets the version of SHIR that is installed on the machine.
+
+    .OUTPUTS
+    SHIR version if SHIR is installed, else $null.
+    #>
+	$InstalledSoftware = Get-ChildItem "HKLM:\\Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\*"
+	foreach ($obj in $InstalledSoftware) {
+		if ($obj.GetValue('DisplayName') -like "*Microsoft Integration Runtime*") {
+			return $obj.GetValue('DisplayVersion')
+		}
+	}
+	return $null
+}
+
+################################### Lock/Unlock Script Instance ##################################
+Function Lock-ScriptInstance {
+	<#
+    .DESCRIPTION
+    The function creates a lock file so that only one instance of the script can run at a time.
+    Return false if already another powershell console has the lock using the script id, else true.
```

Comment:
Please add additional comments on why can't we use the inbuilt Semaphores available in PowerShell

Reply:
Added!